### ISSUE TITLE
References CodeLens for templates

### ISSUE THREAD

**Issue by listepo-alterpost** [angular/vscode-ng-language-service#162]
Hi, can you add References CodeLens when properties or methods using in the templates?

**Comment by webia1** [comment]
https://github.com/microsoft/vscode/issues/80794

**Comment by kyliau** [comment]
Possible duplicate: https://github.com/angular/vscode-ng-language-service/issues/29
What's the difference between "Find all references" and "References CodeLens"?

**Comment by kyliau** [comment]
Will track this in https://github.com/angular/vscode-ng-language-service/issues/29

**Comment by atscott** [comment]
Re-opening since "find all references" is different from CodeLens. Note that the references CodeLens is provided by the built in typescript extension: https://github.com/microsoft/vscode/blob/d99c218e9b2da6935d467a7566a3de49c406ce90/extensions/typescript-language-features/src/languageFeatures/codeLens/referencesCodeLens.ts#L43-L52

We can't "edit" or add template references to those results, but would instead have to provide our own CodeLens for template references.

**Comment by LucasLundJensen** [comment]
Any updates on this?

**Comment by micaminoff** [comment]
I'm refactoring a huge project at the moment. Reliable reference counts would be a major help.

**Comment by Harpush** [comment]
Any news? Adding a # references in template sounds good to me

**Comment by Harpush** [comment]
@atscott it's been so many years? Can this please be implemented?

**Comment by Dozorengel** [comment]
As a workaround I use [TypeLens](https://marketplace.visualstudio.com/items?itemName=kisstkondoros.typelens) right now

**Comment by webia1** [comment]
```plaintext
Dear Guardians of this Code,
Who labor with each byte and node,
Seven years have passed us by,
Since this issue caught the eye.

Like a riddle with no end,
I inquire, my coding friend,
When might thee take a second glance,
And give this lingering quest a chance?

The whiskey's warmth has fueled my pen,
Inspiring me to ask again.

Yours, in lines of code and rhyme,
Awaiting your response in time.
```

**Comment by joeskeen** [comment]
@Dozorengel thanks for the tip about TypeLens! It took a couple minutes after enabling it (probably indexing my project), but now I see my reference from my template!
![image](https://github.com/angular/vscode-ng-language-service/assets/12286274/463fd1c0-0c18-4944-976f-8d9500efe30c)
![image](https://github.com/angular/vscode-ng-language-service/assets/12286274/270a139b-4404-47dc-ba1d-e4586181a11f)

**Comment by Dozorengel** [comment]
@joeskeen yeap, I've been using references all the time since last year. Not all references are displayed, but that's fine enough.

**Comment by joeskeen** [comment]
Yeah, the main one that doesn't get picked up for me are all my `@HostBinding()`s. To change that, the Angular language service would have to report usage of the symbol somewhere. Maybe the @Component() decorator? 🤷

But overall, much better than what I had before!

**Comment by Harpush** [comment]
It will be 7 years tomorrow - any chance it will happen?

**Comment by Harpush** [comment]
This is one of my wishlist issues ☹️
